cranelift/codegen/src/isa/zkasm/inst/mod.rs

```diff
@@ -279,6 +279,18 @@ fn zkasm_get_operands<F: Fn(VReg) -> VReg>(inst: &Inst, collector: &mut OperandC
             collector.reg_clobbers(clobbered);
             collector.reg_def(rd);
         }
+        &Inst::DivArith { rd, rs1, rs2, .. } => {
+            collector.reg_fixed_use(rs1, a0());
+            collector.reg_fixed_use(rs2, b0());
+            let mut clobbered = PRegSet::empty();
+            clobbered.add(a0().to_real_reg().unwrap().into());
+            clobbered.add(c0().to_real_reg().unwrap().into());
+            clobbered.add(d0().to_real_reg().unwrap().into());
+            collector.reg_clobbers(clobbered);
+            // Division via ARITH perform different operations with registers A, B, C, D,
+            // so they can not be result register for proper division
+            collector.reg_fixed_def(rd, e0());
```

Comment:
Huh, I was under impression that in order to check DIV with ARITH, we’d have to use `E`, `D` and `B` registers as inputs, and `A` would be the output or something like that? I’m quite lost how this is supposed to work at this point…

Reply:
Hmm, yes, with convention that div performs ``E / B = A`` we have 2 lines of bytecode less. But we still have to fix all the registers, because if we will allow register allocator freely choose output register, ``B`` for example can be chosen, and it will broke the assertation

Reply:
``D`` and ``C`` also should be set to zero when we perform ``ARITH`` assertation

Reply:
So I think ``A := E / B`` is optimal for ``DivArith`` in assumption ``less lines of zkAsm code => better performance``